cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/MaintenanceTimeWindowsTest.java

```diff
@@ -0,0 +1,308 @@
+
+        Async async = context.async();
+
+        testKafkaRollingUpdate(null,
+            () -> Date.from(LocalDateTime.of(2018, 11, 26, 11, 00, 0).atZone(ZoneId.of("GMT")).toInstant()),
+            r -> {
+                String generation = getClusterCaGenerationPod(KafkaCluster.kafkaPodName(NAME, 0));
+                context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                async.complete();
+            });
+
+        async.await();
+    }
+
+    @Test
+    public void testKafkaRollingUpdateMaintenanceSatisfiedDifferentTz(TestContext context) {
+
+        Async async = context.async();
+
+        testZkRollingUpdate(Collections.singletonList("* * 14-15 * * ?"),
+            () -> Date.from(LocalDateTime.of(2018, 11, 26, 9, 00, 0).atZone(ZoneId.of("Pacific/Easter")).toInstant()),
+            r -> {
+                String generation = getClusterCaGenerationPod(ZookeeperCluster.zookeeperPodName(NAME, 0));
+                context.assertEquals("1", generation, "Pod had unexpected generation " + generation);
+                async.complete();
+            });
+
+        async.await();
+    }
+
+    private String getClusterCaGenerationPod(String podName) {
+        Pod pod = this.mockClient.pods().inNamespace(NAMESPACE).withName(podName).get();
+        return pod.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION);
+    }
+
+    private void testZkRollingUpdate(List<String> maintenanceTimeWindows, Supplier<Date> dateSupplier,
+                                    Handler<AsyncResult<KafkaAssemblyOperator.ReconciliationState>> handler) {
+
+        this.init(maintenanceTimeWindows);
+
+        StatefulSet zkSS = ZookeeperCluster.fromCrd(this.kafka).generateStatefulSet(false);
+        zkSS.getSpec().getTemplate().getMetadata().getAnnotations().put(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "0");
+        this.mockClient.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.zookeeperStatefulSetName(NAME)).create(zkSS);
+
+        this.reconciliationState.zkDiffs = ReconcileResult.created(zkSS);
+        this.reconciliationState.clusterCa = new ClusterCa(null, null, this.clusterCaSecret, null) {
+
+            @Override
+            public boolean certRenewed() {
+                return true;
+            }
+        };
+
+        StatefulSet z = this.mockClient.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.zookeeperStatefulSetName(NAME)).get();
+        z.getSpec().getTemplate().getMetadata().getAnnotations().put(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, "1");
+        this.mockClient.apps().statefulSets().inNamespace(NAMESPACE).withName(KafkaResources.zookeeperStatefulSetName(NAME)).patch(z);
+
+        this.reconciliationState.zkRollingUpdate(dateSupplier).setHandler(handler);
+    }
+
+    private void testKafkaRollingUpdate(List<String> maintenanceTimeWindows, Supplier<Date> dateSupplier,
```

Comment:
Might be handy to document what case this is testing, since it's not so clear from the method name.

Reply:
Actually, it's private and re-used by the real `@Test` methods. What kind of doc should I add here?

Reply:
Well, it doesn't really do a complete reconciliation, so I would be for `doKafkaRollingUpdate` instead of the `test` prefix.

Reply:
Yeah, fine.